### PR TITLE
Use Mono.Cecil instead of System.Reflection

### PR DESCRIPTION
**Pull request by patriksvensson**
Will remove the need for loading/unloading assemblies.

- [x] Script alias finder
- [ ] Default assembly resolver for .NET Core
- [x] Tests from Cake.Core
  - [x] Method alias code generation
  - [x] Property alias code generation
- [x] Method alias code generator
- [x] Property alias code generator
- [x] Documentation support (parsing and CRef generation)
  - [x] Parsing XML
  - [x] CRef generation
  - [x] Method aliases
  - [x] Property aliases

### CODE REVIEW

**Comment by mholo65** [comment]
@patriksvensson this is starting to look really, really good! Great work!

**Comment by mholo65** [comment]
@patriksvensson I will start reviewing shortly.

If ok, could you do `Default assembly resolver for .NET Core` to a separate PR?

**Comment by patriksvensson** [comment]
@mholo65 Yes-a-reno

File: src/Cake.ScriptServer/CodeGen/Generators/CakeMethodAliasGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                    {
+                        writer.WriteLine();
+                        writer.Write("#pragma warning restore 0618");
+                    }
+                }
+            }
+            writer.Write("}");
+        }
+
+        private void WriteInvokation(TextWriter writer, CakeScriptAlias alias)
```
Comment (mholo65) [inline src/Cake.ScriptServer/CodeGen/Generators/CakeMethodAliasGenerator.cs]: WriteInvocation

File: src/Cake.ScriptServer/Reflection/MethodSignature.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+using System.Linq;
+using Cake.ScriptServer.Documentation;
+using Mono.Cecil;
+
+namespace Cake.ScriptServer.Reflection
+{
+    public sealed class MethodSignature
+    {
+        public string CRef { get; }
+        public string Name { get; set; }
```
Comment (mholo65) [inline src/Cake.ScriptServer/Reflection/MethodSignature.cs]: Is `set` needed here?

File: src/Cake.ScriptServer/Reflection/TypeSignature.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+using Mono.Cecil;
+
+namespace Cake.ScriptServer.Reflection
+{
+    public sealed class TypeSignature
+    {
+        public string CRef { get; set; }
+        public string Name { get; set; }
+        public bool IsGenericArgumentType { get; set; }
+        public bool IsArray { get; set; }
```
Comment (mholo65) [inline src/Cake.ScriptServer/Reflection/TypeSignature.cs]: Is all these `setters` needed?

File: src/Cake.ScriptServer/Properties/launchSettings.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "profiles": {
     "Cake.ScriptServer": {
       "commandName": "Project",
-      "commandLineArgs": "\"--assembly=C:\\Users\\mb\\src\\cake-vscode-debug-example\\tools\\Cake\\Cake.Common.dll\""
+      "commandLineArgs": "\"--assembly=C:\\Users\\Patrik\\Desktop\\Cake.Common\\Cake.Common.dll\""
```
Comment (mholo65) [inline src/Cake.ScriptServer/Properties/launchSettings.json]: Should we just ignore this file?

Comment (mholo65) [inline src/Cake.ScriptServer/Properties/launchSettings.json]: https://github.com/github/gitignore/blob/master/VisualStudio.gitignore#L49

File: src/Cake.ScriptServer/CodeGen/Generators/CakeMethodAliasGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+{
+    public sealed class CakeMethodAliasGenerator : CakeAliasGenerator
+    {
+        private readonly TypeEmitter _typeEmitter;
+        private readonly ParameterEmitter _parameterEmitter;
+
+        public CakeMethodAliasGenerator(TypeEmitter typeEmitter, ParameterEmitter parameterEmitter)
+        {
+            _typeEmitter = typeEmitter;
+            _parameterEmitter = parameterEmitter;
```
Comment (mholo65) [inline src/Cake.ScriptServer/CodeGen/Generators/CakeMethodAliasGenerator.cs]: Null checks?
`_parameterEmitter = parameterEmitter ?? throw new ArgumentNullException(nameof(parameterEmitter));`

File: src/Cake.ScriptServer/CodeGen/CakeScriptGenerator.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    internal sealed class CakeScriptGenerator
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly CakeScriptAliasFinder _aliasFinder;
+        private readonly CakeMethodAliasGenerator _methodGenerator;
+        private readonly CakePropertyAliasGenerator _propertyGenerator;
+
+        public CakeScriptGenerator(IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
```
Comment (mholo65) [inline src/Cake.ScriptServer/CodeGen/CakeScriptGenerator.cs]: null check?

**Pull request merged**